src/libraries/System.Runtime/tests/System.Globalization.Tests/CompareInfo/CompareInfoTests.HashCode.cs

```diff
@@ -74,42 +74,47 @@ private void TryAddToCustomDictionary(StringComparer comparer, string str1, stri
-            yield return new object[] { '\u2013', thaiCmpInfo }; // DashPunctuation: EN DASH
-            yield return new object[] { '\u005F', thaiCmpInfo }; // ConnectorPunctuation: LOW LINE
-            yield return new object[] { '\u2018', thaiCmpInfo }; // InitialQuotePunctuation: LEFT SINGLE QUOTATION MARK
-            yield return new object[] { '\u2019', thaiCmpInfo }; // FinalQuotePunctuation: RIGHT SINGLE QUOTATION MARK
-            yield return new object[] { '\u2028', thaiCmpInfo }; // LineSeparator: LINE SEPARATOR
-            yield return new object[] { '\u2029', thaiCmpInfo }; // ParagraphSeparator: PARAGRAPH SEPARATOR
+            string str1 = "ab";
+            // browser supports 240 cultures (e.g. "en-US"), out of which 54 neutral cultures (e.g. "en")
+            CultureInfo[] cultures = CultureInfo.GetCultures(CultureTypes.NeutralCultures);
+            List<int> ignoredCodepoints = new();
+            foreach(var culture in cultures)
+            {
+                // japanese with None is not supported, JS always ignores Kana when localeCompare is used
+                CompareOptions options = culture.Name == "ja" ?
+                    CompareOptions.IgnoreKanaType :
+                    CompareOptions.None;
+                CompareInfo cmpInfo = culture.CompareInfo;
+                var hashCode1 = cmpInfo.GetHashCode(str1, options);
+                for(int codePoint = 0; codePoint < 0x10FFFF; codePoint++)
+                {
+                    char character = (char)codePoint;
+					string str2 = $"a{character}b";
+                    // in HybridGlobalization CompareInfo.Compare uses JS's localeCompare()
+                    if (cmpInfo.Compare(str1, str2, options) == 0)
+                    {
+                        // do not test same codepoint with different cultures
+                        if (!ignoredCodepoints.Contains(codePoint))
+                        {
+                            ignoredCodepoints.Add(codePoint);
+                            yield return new object[] { hashCode1, str2, cmpInfo, options };
+                        }
+                    }
+                }
+            }
         }
 
-        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsIcuGlobalization))]
-        [MemberData(nameof(CheckHashingOfSkippedChars_TestData))]
-        public void CheckHashingOfSkippedChars(char character, CompareInfo cmpInfo)
+        // In non-hybrid we have hashing and Equal function from the same source - ICU4C, so this test is not necessary
+        // Hybrid has Equal function from JS and hashing from managed invariant algorithm, they might start diverging at some point
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsHybridGlobalizationOnBrowser))]
+        [MemberData(nameof(CharsIgnoredByEqualFunction))]
+        public void CheckHashingOfSkippedChars(int hashCode1, string str2, CompareInfo cmpInfo, CompareOptions options)
```

Comment:
I think it would be good to also have another test like

```
foreach locale
    foreach codepoint
        var s1=$"A{codepoint}B"
        var s2=$"AB"
        var h1 = locale.getHash(s1)
        var h2 = locale.getHash(s2)
        if(locale.equals(s1, s2))
             assert(h1 == h2)
        else
             // We know that the hash collisions are OK, when they are rare. So this should fail in very small % of cases.
             assert(h1 != h2)
```

After we learned how bad the hash collisions are, we could comment out `assert(h1 != h2)` or add few known collisions as exception to the rules.